ui/src/app/workflows/components/workflow-timeline/workflow-timeline.tsx

```diff
@@ -18,128 +18,124 @@ interface WorkflowTimelineProps {
     nodeClicked?: (node: models.NodeStatus) => any;
 }
 
-interface WorkflowTimelineState {
-    parentWidth: number;
-    now: moment.Moment;
-}
+export function WorkflowTimeline(props: WorkflowTimelineProps) {
+    let resizeSubscription: Subscription;
+    let refreshSubscription: Subscription;
 
-export class WorkflowTimeline extends React.Component<WorkflowTimelineProps, WorkflowTimelineState> {
-    private container: HTMLElement;
-    private resizeSubscription: Subscription;
-    private refreshSubscription: Subscription;
+    const [parentWidth, setParentWidth] = useState(0);
+    const [now, setNow] = useState(moment());
 
-    constructor(props: WorkflowTimelineProps) {
-        super(props);
-        this.state = {parentWidth: 0, now: moment()};
-        this.ensureRunningWorkflowRefreshing(props.workflow);
-    }
+    const containerRef = useRef(null);
```

Comment:
```suggestion
    const containerRef = useRef<HTMLElement>(null);
```

this one _does_ need a type annotation 😉 
(since `null` can't be inferred to be an `HTMLElement`)